var/spack/repos/builtin/packages/heffte/package.py

```diff
@@ -146,17 +126,19 @@ def test_make_test(self):
         cmake_dir = self.test_suite.current_test_cache_dir.testing
 
         options = [cmake_dir]
+        options.append(f"-DHeffte_DIR={self.spec.prefix.lib.cmake.Heffte}")
         if "+rocm" in self.spec:
+            # path name is 'hsa-runtime64' but python cannot have '-' in variable name
+            hsa_runtime = join_path(self.spec["hsa-rocr-dev"].prefix.lib.cmake, "hsa-runtime64")
             options.extend(
                 [
                     f"-Dhip_DIR={self.spec['hip'].prefix.lib.cmake.hip}",
                     "-DAMDDeviceLibs_DIR="
                     + f"{self.spec['llvm-amdgpu'].prefix.lib.cmake.AMDDeviceLibs}",
                     f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}",
-                    "-Dhsa-runtime64_DIR="
-                    + f"{self.spec['hsa-rocr-dev'].prefix.lib.cmake.hsa-runtime64}",
-                    "-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}",
-                    "-Drocfft_DIR={self.spec['rocfft'].prefix.lib.cmake.rocfft}",
+                    "-Dhsa-runtime64_DIR=" + hsa_runtime,
+                    f"-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}",
+                    f"-Drocfft_DIR={self.spec['rocfft'].prefix.lib.cmake.rocfft}",
```

Comment:
```suggestion
                    self.define("hsa-runtime64_DIR", hsa_runtime),
                    self.define("HSA_HEADER", self.spec['hsa-rocr-dev'].prefix.include),
                    self.define("rocfft_DIR", self.spec['rocfft'].prefix.lib.cmake.rocfft),
```

We should update these calls to use `self.define(key, value)` instead of format strings.

Reply:
I updated the syntax